### PR TITLE
feat(dart_frog_cli): add `dart_frog daemon` command

### PR DESCRIPTION
**Pull request by renancaraujo**
## Status

**READY**

## Description

feat: add `dart_frog daemon` command

Adding this command with basic placeholder behavior. 

The first step into #705 

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/dart_frog_cli/test/src/commands/daemon/daemon_test.dart [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        logger: logger,
+        pubUpdater: _MockPubUpdater(),
+        exit: (_) {},
+        sigint: sigint,
+      );
+    });
+
+    test('should instantiate and run with default value', () async {
+      final DaemonCommand command;
+      expect(command = DaemonCommand(), isNotNull);
```
Comment (renancaraujo) [inline packages/dart_frog_cli/test/src/commands/daemon/daemon_test.dart:52]: Trash from previous implementation, you (I) should remove it.

File: packages/dart_frog_cli/lib/src/commands/daemon/daemon.dart [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  ///
+  /// Visible for testing purposes only.
+  @visibleForTesting
+  Daemon get daemon => _daemonBuilder(logger);
+
+  @override
+  Future<int> run() async {
+    final exit = await daemon.exitCode;
+    return exit.code;
+  }
```
Comment (erickzanardo) [inline packages/dart_frog_cli/lib/src/commands/daemon/daemon.dart:43]:
```suggestion
  Future<int> run() => daemon.exitCode;
```

File: packages/dart_frog_cli/lib/src/daemon/daemon.dart [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      (_) {
+        _logger.detail('Killing Dart Frog daemon...');
+        kill(ExitCode.success);
+      },
+    );
+  }
+
+  final Logger _logger;
+
+  final Completer<ExitCode> _exitCodeCompleter = Completer<ExitCode>();
```
Comment (erickzanardo) [inline packages/dart_frog_cli/lib/src/daemon/daemon.dart:28]:
```suggestion
  final  _exitCodeCompleter = Completer<ExitCode>();
```

**Review by erickzanardo — DISMISSED** [review]
couple of small comments, but LGTM!

**Review by alestiago — APPROVED** [review]
Excited to see this become a thing 🥳

**Pull request merged**